src/shortestpaths/desopo-pape.jl

```diff
@@ -0,0 +1,46 @@
+"""
+    struct DEposoPapeState{T, U}
+
+An [`AbstractPathState`](@ref) designed for D`Esopo-Pape shortest-path calculations.
+"""
+struct DEsopoPapeState{T <:Real, U <: Integer} <: AbstractPathState
+    parents::Vector{U}
+    dists::Vector{T}
+end
+
+function DEsopoPape_shortest_path(g::AbstractGraph, 
+    src::U,
+    distmx::AbstractMatrix{T} = weights(g)) where T <: Real where U <: Integer
+    
+    nvg = nv(g)
+    dists = fill(typemax(T), nvg)
+    parents = zeros(U, nvg)
+    state = fill(2, nvg)
+    q = Array{U,1}()
+    push!(q, src)
+    dists[src] = 0
+    
+    while !isempty(q)
```

Comment:
You have a lot of array access here, so you can use `@inbounds` here.

Reply:
I was skeptical initially about this. Is this necessary?

Reply:
No, but it can make stuff much faster, because otherwise you would have a branch for every array access. Might be a good idea to only add `@inbounds`, when you are finished with writing an algorithm, so until then, you see when you make an error.